src/util/util.h

```diff
@@ -173,10 +173,15 @@ namespace Util
 	void hatchingOperation(const QRectF& extent, double spacing, double offset, double rotation, T& processor)
 	{
 		// Make rotation unique
-		rotation = fmod(1.0 * rotation, M_PI);
+		rotation = fmod(1.0 * rotation, 2 * M_PI);
 		if (rotation < 0)
-			rotation = M_PI + rotation;
-		Q_ASSERT(rotation >= 0 && rotation <= M_PI);
+			rotation = 2 * M_PI + rotation;
+		Q_ASSERT(rotation >= 0 && rotation <= 2 * M_PI);
+		while (rotation > M_PI - 0.0001)
```

Comment:
I see how it works, but I don't feel like `while` is the best expression here.
- The loop can run exactly zero, one, or two times, given the previous assertion.
- When running two times, it will move rotation out of the previously asserted range.

IIUC this loop can be removed when an extra `|| qAbs(rotation > M_PI - 0.0001)` is added in line 195?

Reply:
Okay, updated the comment twice, but still it is not correct: Instead of the `while` loop, it still takes an `if`:
~~~
		if (rotation >= M_PI)
		{
			rotation -= M_PI;
			offset = -offset;
		}
		Q_ASSERT(rotation >= 0 && rotation < M_PI);
~~~
and later for horizontal lines:
~~~
		else if (rotation < 0.0001 || rotation > M_PI - 0.0001)

Reply:
@dg0yt Yes, the `while` was tricky. It would be more straightforward to do it as you say. With the 
`		else if (rotation < 0.0001 || rotation > M_PI - 0.0001)`
that you suggest, it would require another
`		if (rotation > M_PI / 2) offset = -offset;`
which is what the trickiness avoided. Straightforward is good.

Reply:
Done.